File: ww/src/main/java/app/agentundercover/AgentUndercoverService.java
package app.agentundercover;

import java.util.List;

import org.springframework.stereotype.Service;

import lombok.Getter;
import model.agentundercover.Daten;
import model.agentundercover.Spieler;

@Service
@Getter
public class AgentUndercoverService {

    private Daten daten = new Daten();

    public void setSpieler(List<Spieler> spieler) {

        daten.setSpieler(spieler);
    }

    public void registriere(String name) {

        daten.registriere(name);
    }

    public void bestaetigeOrt(String name) {

        daten.bestaetigeOrt(name);
    }

    public void neustart() {

        daten.neustart();
    }
}

File: ww/web/src/app/agentundercover.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";


@Injectable()
export class AgentUndercoverService {

    private url = '/api/agentundercover';

    constructor(
        private http: HttpClient
    ) { }

    public getDaten(): Observable<any> {

        return this.http.get<any>(`${this.url}/daten`);
    }

    public setSpieler(spieler: any) {

        return this.http.post(`${this.url}/spieler`, spieler);
    }

    public registriere(name: string) {

        return this.http.post(`${this.url}/registriere`, name);
    }

    public bestaetigeOrt(name: string) {

        return this.http.post(`${this.url}/bestaetigeOrt`, name);
    }

    public neustart() {

        return this.http.get(`${this.url}/neustart`);
    }
}